src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,737 @@
+    const sleep = milliseconds => {
+      return new Promise(resolve => setTimeout(resolve, milliseconds));
+    };
+
+    beforeEach(() => {
+      api = http2.createServer();
+      api.on("error", err => console.error(err));
+
+      api.on("stream", async (stream, headers) => {
+        const path = headers[":path"];
+        const method = headers[":method"];
+
+        if (method !== "GET") {
+          stream.respond({
+            ":status": 404
+          });
+          stream.end();
+        }
+        switch (path) {
+          case "/version?package=ganache":
+            if (stream.closed) return;
+            stream.respond({
+              ":status": 200
+            });
+            stream.write(apiResponse);
+            stream.end();
+            break;
+          default:
+            await sleep(10);
+            stream.respond({
+              ":status": 404
+            });
+            stream.end();
+            break;
+        }
+      });
+
+      api.listen(apiSettings.port);
+      vc = new VersionCheck(
+        testVersion,
+        {
+          enabled: true,
+          url: "http://localhost:" + apiSettings.port
+        },
+        testLogger
+      );
+    });
+
+    afterEach(() => {
+      if (api) {
+        api.close();
+      }
+    });
+    it("will not getLatestVersion if version check is disabled", async () => {
+      vc.disable();
+
+      await vc.getLatestVersion();
+
+      const { latestVersion } = vc.getConfig();
+
+      assert.strictEqual(
+        latestVersion,
+        testVersion,
+        "Version Check will _fetchLatest if disabled."
+      );
+    });
+
+    it("fetches the latest version from the API", async () => {
+      let latestVersion;
+
+      latestVersion = await vc._fetchLatest();
+
+      assert.strictEqual(
+        latestVersion === apiResponse,
+        true,
+        "latestVersion does not match apiResponse."
+      );
+    });
+    /*
+    This will only ever cause intermittent failures in mocha on mac.
+    it("When the ttl expires, throw and cancel the request", async () => {
+      const ttl = 1;
+
+      vc = new VersionCheck(
+        testVersion,
+        {
+          enabled: true,
+          url: "http://localhost:" + apiSettings.port,
+          packageName: "slow",
+          ttl
+        },
+        testLogger
+      );
+
+      let errorMessage;
+      try {
+        await vc._fetchLatest();
+      } catch (e) {
+        errorMessage = e;
+      }
+
+      assert.strictEqual(errorMessage, `ttl expired: ${ttl}`);
+    });
+*/
+    it("fetches the latest version and sets it in the config file.", async () => {
+      await vc.getLatestVersion();
+
+      const latestVersion = vc._config.latestVersion;
+
+      assert.strictEqual(
+        latestVersion,
+        apiResponse,
+        "latestVersion is not correctly set in _config after getLatestVersion."
+      );
+    });
+  });
+
+  describe("destroy", () => {
+    it("sets status to 'destroyed'", () => {
```

Comment:
it's actually just asserting that the _request property is set to null right?

Reply:
yus.